### PR TITLE
Closes #207 xportr_length for missing length in metadata 

### LINKED ISSUE #207: xportr_length: Set ---DTC variables length to maximum length based Date (10) or Date/Time (19)
Update to `xportr_length()` that sets DTC variables to maximum length based on Date (10) or Date/Time (19) inputs.  

Add a check/warning when the spec length is populated for the --DTC variables as the define.xml should not have a length (missing).

### PR DESCRIPTION
**Pull request by cpiraux**
### Thank you for your Pull Request!

We have developed a Pull Request template to aid you and our reviewers. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs. We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

### Changes Description

Assigns the maximum length value instead of 200 for a character variable when the length is missing in the metadata

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description. Can be removed or left blank if changes are minor/self-explanatory.
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Use `styler` package and functions to style files accordingly.
- [x] New functions or arguments follow established convention found in the [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers).
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] The `NEWS.md` entry should go under the `# xportr development version` section. Don't worry about updating the version because it will be auto-updated using the `vbump.yaml` CI.
- [x] Address any updates needed for vignettes and/or templates.
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] The developer is responsible for fixing merge conflicts not the Reviewer.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-metadata.R [hunk 1]
```diff
@@ -513,12 +513,12 @@ test_that("xportr_length: Check if length gets imputed when a new variable is pa
     xportr_length(df, df_meta, domain = "df")
   )
 
-  # 200 is the imputed length for character and 8 for other data types as in impute_length()
-  expect_equal(c(x = 1, y = 200, z = 8), map_dbl(df_with_width, attr, "width"))
+  # Max length is the imputed length for character and 8 for other data types as in impute_length()
```
Comment (bms63) [inline tests/testthat/test-metadata.R:516]: impute_length() was removed.  Can this test subject header be updated as well.

**Review by bms63 — CHANGES_REQUESTED** [review]
Just a small update to test and please fix lintr issue.  Thanks again for this work @cpiraux !!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by cpiraux** [comment]
What should I do to resolve this lint error?
image

**Comment by bms63** [comment]
> What should I do to resolve this lint error? image

It has to do with the number of conditions being used in a function.  https://en.wikipedia.org/wiki/Cyclomatic_complexity

I increased it to 18 in the `.lintr` file so this would check.  Will make an issue for someone to look into for next release.

**Review by bms63 — APPROVED** [review]

**Pull request merged**